### DISCUSSION TITLE
Modern Site Pages migration with Get-PnPProvisioningTemplate

### DISCUSSION THREAD

**Discussion by orangejuicejones** [pnp/powershell#120] [General] [answered]
Acording to this article, I understand that pnp can now migrate pages.
https://sharepoint.handsontek.net/2019/06/16/how-to-migrate-content-with-pnp-provisioning-engine/

So I executed the below commands:
[old site]
Get-PnPProvisioningTemplate -Out CS_T02_test_template.xml  -Handlers PageContents -IncludeAllClientSidePages
[new site]
Apply-PnPProvisioningTemplate -Path CS_T02_test_template.xml

But get error:
_Apply-PnPProvisioningTemplate : Value cannot be null.
Parameter name: stream
At line:1 char:1
+ Apply-PnPProvisioningTemplate -Path CS_T02_test_template.xml
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPProvisioningTemplate], ArgumentNullException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.ApplyProvisioningTemplate_

SharePointPnPPowerShellOnline 3.28.2012.0

**Comment by erwinvanhunen** [comment, marked as answer]
By default when we extract a template for a site we include references for a file, but do not download the actual file. The error message you're seeing is that the engine wants to provision a file, but it cannot find it on the file system where the template resides. If you include the -PersistBrandingFiles parameter you will notice a SiteAssets folder being created at the current location where you execute the cmdlet with in there the files. The template will refer to those files and will be able to find them to upload them.
